src/Core/Test.php

```diff
@@ -52,13 +54,21 @@ public function run(TestResult $result)
     protected function executeTest(TestResult $result)
     {
         $action = ['passTest', $this];
+        $handlers = $this->handleErrors($result, $action);
         try {
             $this->runSetup();
             call_user_func_array($this->getDefinition(), $this->getDefinitionArguments());
+        } catch (Throwable $e) {
+            if ('passTest' === $action[0]) {
```

Comment:
This pattern seems to be repeated in a few places, any way we can pull this out into a common method/function? Does it seem worth it?

Something like `failIfPassing(array $action): array`?